### ISSUE TITLE
compiling with features?

### ISSUE THREAD

**Issue by fdncred** [sourcegear/rust-vs-extension#6]
I have a project that I want to debug but I need to compile it with `--features=extra` when I hit F5. How do I do that?

**Comment by ericsink** [comment]
That's not supported yet. We've got a bunch of those things that need to be done, and we're trying to figure out the best way to implement them.

**Comment by ericsink** [comment]
Version 0.6 contains an undocumented and temporary fix for this problem.

Add a text file called `unstable_cargo_features.txt` as a sibling to `Cargo.toml`.  In that file, put the names of the features you want specified, one per line.

This is just a temporary workaround until we expose a proper way to do this in the UI.

**Comment by ericsink** [comment]
Darn. You're not doing anything wrong. I messed this up.

I actually coded two temporary workarounds.  The `unstable_cargo_features.txt` approach is used for Build All.

For building for debug launch, I started with a different approach, and I forgot to change over to the `unstable_cargo_features.txt` way.

But the original approach is still there, so if you want to try it:

Add a text file called `VSWorkspaceSettings.json` as a sibling to `Cargo.toml`.  The structure of that JSON file looks like this:

```
{
    "package_name" :
    {
        "target_name" :
        {
            "features" :
            [
                "feature_name",
                "another_feature_name"
            ]
        }
    }
}
```

So for example, if your package and target are both named "foo", and what you want is `--features=extra`, the JSON would look like this:

```
{
    "foo" :
    {
        "foo" :
        {
            "features" :
            [
                "extra"
            ]
        }
    }
}
```

This too is just a temporary workaround.

My apologies.
